noteable_magics/sql/meta_commands.py

```diff
@@ -543,53 +544,57 @@ def run(self, invoked_as: str, args: List[str]) -> None:
             # Cannot continue, in that there's no place to store the results gate-side, since there's
             # no corresponding datasources row to relate to.
 
+            # This will not happen in modern in any headless introspections -- both Gate and Geas
+            # protect against headlessly introspecting the legacy datasources.
+
             print('Cannot introspect into this resource.', file=sys.stderr)
             return (None, False)
 
-        inspector = self.get_inspector()
+        # RelationStructureMessager handles both:
+        #
+        #   * Uploading batches of successfully discovered relations
+        #   * Catching any exception and reporting it back to gate. Will suppress the exception.
+        #
+        with RelationStructureMessager(ds_id) as messenger:
```

Comment:
Use the RelationStructureMessager context manager earlier, so can capture and report any exceptions encountered when trying to connect to the database within the get_inspector() call.